WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanViewModel.kt

```diff
@@ -1,15 +1,65 @@
 package org.wordpress.android.ui.jetpack.scan
 
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModel
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.scan.ScanStateModel
+import org.wordpress.android.fluxc.model.scan.ScanStateModel.State
+import org.wordpress.android.ui.jetpack.scan.ScanListItemState.ScanState.ScanIdleState
+import org.wordpress.android.ui.jetpack.scan.ScanListItemState.ScanState.ScanScanningState
+import org.wordpress.android.ui.jetpack.scan.ScanViewModel.UiState.Content
 import javax.inject.Inject
 
-class ScanViewModel @Inject constructor() : ViewModel() {
+class ScanViewModel @Inject constructor(
+    private val scanStatusService: ScanStatusService
+) : ViewModel() {
     private var isStarted = false
+    private var lastScanState: State? = null
 
-    fun start() {
+    private val _uiState: MutableLiveData<UiState> = MutableLiveData()
+    val uiState: LiveData<UiState> = _uiState
+
+    private val scanStateObserver = Observer<ScanStateModel> {
+        if (it.state != lastScanState) {
+            lastScanState = it.state
+            _uiState.value = buildContentUiState(it)
+        }
+    }
+
+    lateinit var site: SiteModel
+
+    fun start(site: SiteModel) {
         if (isStarted) {
             return
         }
```

Comment:
Minor (🔍 ): Consider keeping the `if (isStarted)` close to the `isStated = true` logic. Otherwise this can have the potential to become a problem in the future.
Minor (🔍 ): Consider have the `if (isStarted) return` a one-liner. But, as I said, this is just what I would do.

Reply:
> Consider keeping the if (isStarted) close to the isStated = true logic. Otherwise this can have the potential to become a problem in the future.

👍

> Consider have the if (isStarted) return a one-liner. But, as I said, this is just what I would do.

Followed existing code conventions which uses braces around if:

```
if (isStarted) {
    return
}
```